java/core/src/main/java/com/whylogs/core/metrics/Metric.java

```diff
@@ -0,0 +1,48 @@
+package com.whylogs.core.metrics;
+
+import com.whylogs.core.PreProcessedColumn;
+import com.whylogs.core.SummaryConfig;
+
+import java.util.HashMap;
+
+public abstract class Metric{
+
+    private String namespace;
+
+    // TODO: needs a __post_init__ for the registry
+
+    public static String getNameSpace(Metric metric, MetricConfig config){
+        return metric.zero(config).namespace;
+    }
+
+    public abstract <T extends Metric> T zero(MetricConfig config);
+
+    public <T extends Metric> T add(Class<T> otherMetric){
```

Comment:
Another weird one... the above says `add()` returns the same class as `otherMetric`, but what we want to say is that `otherMetric` and the return value are the same class as `this`. I'd probably just use `Metric`, assertions, and comments :)
Or maybe make it a ctor and assert the class of `otherMetric` is the same as `this`'.

Reply:
hmm I agree that this won't be the correct situation, but I think I have a different way of doing it that will allow not need the assertions